src/scenes/MainScene.ts

```diff
@@ -0,0 +1,48 @@
+import Phaser from 'phaser';
+import { generateHills } from '@src/map/generateHills';
+import Ball from '@src/entities/Ball';
+import Goal from '@src/entities/Goal';
+import Ground from '@src/entities/Ground';
+
+export default class MainScene extends Phaser.Scene {
+  goal?: Goal;
+  ball?: Ball;
+  ground?: Ground;
+
+  constructor() {
+    super({ key: 'MainScene' });
+  }
+
+  preload() {
+    this.load.image('goal', 'assets/goalflag.png');
+  }
+
+  create() {
+    const hills = generateHills(
+      this.game.config.width as number,
+      this.game.config.height as number
+    );
+
+    this.ground = new Ground(this, hills);
+    this.ball = new Ball(this);
+    this.goal = new Goal(this);
+
+    // eslint-disable-next-line @typescript-eslint/ban-ts-comment
+    // @ts-ignore
+    this.ball.setOnCollideWith(this.goal, () => {
+      this.endGame();
+    });
+
+    // eslint-disable-next-line @typescript-eslint/ban-ts-comment
+    // @ts-ignore
+    this.goal.setOnCollideWith(this.matter.world.getAllBodies(), () => {
+      // eslint-disable-next-line @typescript-eslint/ban-ts-comment
+      // @ts-ignore
+      this.goal.body.isStatic = true;
+    });
```

Comment:
Okay I spent a few hours checking why this is so weird and I think I got an okayish solution that I will push.

Interestingly, the MatterJS physics engine we are using is normally a completely separate physics library. This seems to just be stacked on top of Phaser3 which may be the reason for the wacky typings

Reply:
Your hours spend are greatly appreciated, I was loosing hope and started looking into excalibur.js
with your adjustments this looks way prettier and manageable